### ISSUE TITLE
GUI error in player lists

### ISSUE THREAD

**Issue by derjungeinhd** [WasabiThumb/xclaim#71]
Everytime I try to add a player to a claim or try to rename a claim, the GUI just closes. There is no error massage in the chat or console at all.

I had this Issue on a Paper 1.20.4 Server with XClaim 1.11.0.

Here is a short clip of what happens:
![issue](https://github.com/WasabiThumb/xclaim/assets/125304376/efc1716f-eea4-41ed-8467-8da65204a733)

Here a little longer clip:
https://github.com/WasabiThumb/xclaim/assets/125304376/979db6b9-2c5d-4f8f-b552-1a93e163130d

**Comment by WasabiThumb** [comment]
You are meant to type the text in chat.

**Comment by derjungeinhd** [comment]
So, this is interesting. I downloaded now version 1.12.0 of xclaim and the Chat still doesn't automaticly show up. But more interesting is, that the next thing you type in the Chat will actually be catched by the plugin and then it opens up an empty Xclaim Config menu. Even more wierdly, the Edit Trusted Players Menu is now also completely empty but I can still click the Invisible Options I had before.

**Comment by WasabiThumb** [comment]
Do you get an error with that?

**Comment by derjungeinhd** [comment]
Ok, I get an Error when I open up the Edit Trusted Players Menu. I still need to test the rename claim. This is what the Error says: 
[09:42:09 ERROR]: Could not pass event InventoryClickEvent to XClaim v1.12.0
java.lang.NullPointerException: Profile name must not be null
        at java.util.Objects.requireNonNull(Objects.java:235) ~[?:?]
        at com.mojang.authlib.GameProfile.<init>(GameProfile.java:31) ~[authlib-6.0.52.jar:?]
        at org.bukkit.craftbukkit.v1_20_R3.inventory.CraftMetaSkull.setOwningPlayer(CraftMetaSkull.java:246) ~[paper-1.20.4.jar:git-Paper-446]
        at codes.wasabi.xclaim.platform.spigot_1_12_2.SpigotPlatform_1_12_2.setOwningPlayer(SpigotPlatform_1_12_2.java:13) ~[xclaim-1.12.0.jar:?]
        at codes.wasabi.xclaim.gui.page.PlayerCombinatorPage.populate(PlayerCombinatorPage.java:85) ~[xclaim-1.12.0.jar:?]
        at codes.wasabi.xclaim.gui.page.PlayerCombinatorPage.onEnter(PlayerCombinatorPage.java:102) ~[xclaim-1.12.0.jar:?]
        at codes.wasabi.xclaim.gui.GUIHandler.switchPage(GUIHandler.java:87) ~[xclaim-1.12.0.jar:?]
        at codes.wasabi.xclaim.gui.Page.switchPage(Page.java:85) ~[xclaim-1.12.0.jar:?]
        at codes.wasabi.xclaim.gui.page.MainPage.onClick(MainPage.java:90) ~[xclaim-1.12.0.jar:?]
        at codes.wasabi.xclaim.gui.GUIHandler.onClick(GUIHandler.java:141) ~[xclaim-1.12.0.jar:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor183.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:?]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:81) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:git-Paper-446]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-1.20.4.jar:git-Paper-446]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[paper-1.20.4.jar:git-Paper-446]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerClick(ServerGamePacketListenerImpl.java:3106) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:58) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:23) ~[?:?]
        at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:54) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1455) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:194) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1432) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1355) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
        at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1333) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1222) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[paper-1.20.4.jar:git-Paper-446]
        at java.lang.Thread.run(Thread.java:840) ~[?:?]

**Comment by WasabiThumb** [comment]
> Ok, I get an Error when I open up the Edit Trusted Players Menu. I still need to test the rename claim. This is what the Error says: 
> [09:42:09 ERROR]: Could not pass event InventoryClickEvent to XClaim v1.12.0
> java.lang.NullPointerException: Profile name must not be null
>         at java.util.Objects.requireNonNull(Objects.java:235) ~[?:?]
>         at com.mojang.authlib.GameProfile.<init>(GameProfile.java:31) ~[authlib-6.0.52.jar:?]
>         at org.bukkit.craftbukkit.v1_20_R3.inventory.CraftMetaSkull.setOwningPlayer(CraftMetaSkull.java:246) ~[paper-1.20.4.jar:git-Paper-446]
>         at codes.wasabi.xclaim.platform.spigot_1_12_2.SpigotPlatform_1_12_2.setOwningPlayer(SpigotPlatform_1_12_2.java:13) ~[xclaim-1.12.0.jar:?]
>         at codes.wasabi.xclaim.gui.page.PlayerCombinatorPage.populate(PlayerCombinatorPage.java:85) ~[xclaim-1.12.0.jar:?]
>         at codes.wasabi.xclaim.gui.page.PlayerCombinatorPage.onEnter(PlayerCombinatorPage.java:102) ~[xclaim-1.12.0.jar:?]
>         at codes.wasabi.xclaim.gui.GUIHandler.switchPage(GUIHandler.java:87) ~[xclaim-1.12.0.jar:?]
>         at codes.wasabi.xclaim.gui.Page.switchPage(Page.java:85) ~[xclaim-1.12.0.jar:?]
>         at codes.wasabi.xclaim.gui.page.MainPage.onClick(MainPage.java:90) ~[xclaim-1.12.0.jar:?]
>         at codes.wasabi.xclaim.gui.GUIHandler.onClick(GUIHandler.java:141) ~[xclaim-1.12.0.jar:?]
>         at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor183.execute(Unknown Source) ~[?:?]
>         at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:?]
>         at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:81) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:git-Paper-446]
>         at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:?]
>         at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-1.20.4.jar:git-Paper-446]
>         at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[paper-1.20.4.jar:git-Paper-446]
>         at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:?]
>         at net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerClick(ServerGamePacketListenerImpl.java:3106) ~[?:?]
>         at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:58) ~[?:?]
>         at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:23) ~[?:?]
>         at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:54) ~[?:?]
>         at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.4.jar:git-Paper-446]
>         at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
>         at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
>         at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1455) ~[paper-1.20.4.jar:git-Paper-446]
>         at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:194) ~[paper-1.20.4.jar:git-Paper-446]
>         at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
>         at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1432) ~[paper-1.20.4.jar:git-Paper-446]
>         at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1355) ~[paper-1.20.4.jar:git-Paper-446]
>         at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
>         at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1333) ~[paper-1.20.4.jar:git-Paper-446]
>         at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1222) ~[paper-1.20.4.jar:git-Paper-446]
>         at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[paper-1.20.4.jar:git-Paper-446]
>         at java.lang.Thread.run(Thread.java:840) ~[?:?]

It's using the wrong platform (SpigotPlatform_1_12_2) for some reason. What is the output of /xc info? Are you running a normal Paper 1.20.4 server or some kind of fork like Purpur/Pufferfish?

**Comment by derjungeinhd** [comment]
I don't get an Error by renaming a claim, but the Chat doesn't automaticly open, if that was intended. When I type something in the chat, the claim was renamed successfully and the xclaim menu opens up again. But there is no message that it worked tho.
My Server is running on a normal Paper 1.20.4 installation, but /xc info says the API version is 1.13. Maybe this causes the bugs?

**Comment by derjungeinhd** [comment]
Back to the Trusted Players, after I type something in the Chat to add a Player, this error shows up: [09:43:10 WARN]: [XC] Task #81 for XClaim v1.12.0 generated an exception
java.lang.NullPointerException: Profile name must not be null
        at java.util.Objects.requireNonNull(Objects.java:235) ~[?:?]
        at com.mojang.authlib.GameProfile.<init>(GameProfile.java:31) ~[authlib-6.0.52.jar:?]
        at org.bukkit.craftbukkit.v1_20_R3.inventory.CraftMetaSkull.setOwningPlayer(CraftMetaSkull.java:246) ~[paper-1.20.4.jar:git-Paper-446]
        at codes.wasabi.xclaim.platform.spigot_1_12_2.SpigotPlatform_1_12_2.setOwningPlayer(SpigotPlatform_1_12_2.java:13) ~[xclaim-1.12.0.jar:?]
        at codes.wasabi.xclaim.gui.page.PlayerCombinatorPage.populate(PlayerCombinatorPage.java:85) ~[xclaim-1.12.0.jar:?]
        at codes.wasabi.xclaim.gui.page.PlayerCombinatorPage.lambda$onClick$1(PlayerCombinatorPage.java:128) ~[xclaim-1.12.0.jar:?]
        at codes.wasabi.xclaim.gui.Page.lambda$prompt$1(Page.java:72) ~[xclaim-1.12.0.jar:?]
        at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.20.4.jar:git-Paper-446]
        at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:482) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1636) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:447) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1515) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1216) ~[paper-1.20.4.jar:git-Paper-446]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[paper-1.20.4.jar:git-Paper-446]
        at java.lang.Thread.run(Thread.java:840) ~[?:?]

**Comment by WasabiThumb** [comment]
Yes, that's the same error again. I'm not sure why, but the only way this could happen is if PaperLib is adamant that you are either:
- Not running a Paper server
- Not running at least 1.17
And to that I say: 🤷

**Comment by WasabiThumb** [comment]
Ok. I see what happened now. [This PR](https://github.com/WasabiThumb/xclaim/pull/67) actually had some fairly major issues with it. Last time I'm accepting code changes from others without testing extensively smh. What I said would have been true if the code was how I last remembered it being.

**Comment by derjungeinhd** [comment]
Ok, good to know. Maybe a Chat message of some sorts that you need to open chat and type in it would be helpful. And also a message if there was something wrong with the input or not.

**Comment by WasabiThumb** [comment]
I can't reproduce your exact issue on my test server, but I have a potential fix on [this branch](https://github.com/WasabiThumb/xclaim/tree/bug/skull-owner).
Here's a JAR if you don't want to build it from source:
[Torrent](magnet:?xt=urn:btih:XWQDZUNPOEJG6UP3TAQ4LHAB6HTTKTCN&dn=xclaim-1.12.0-i71.jar&xl=3543556&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337%2Fannounce)
[MEGA](https://mega.nz/file/dzQnVKSQ#q-7mI9fTocWm3ZVnEhKGvAU22b4-Hf0ZYNNcx5lpyEU)

**Comment by derjungeinhd** [comment]
Hey, I would like to build it myself. Do you have a guide how to do it? Sorry, I don't have much experience with plugin development...

**Comment by WasabiThumb** [comment]
> Hey, I would like to build it myself. Do you have a guide how to do it? Sorry, I don't have much experience with plugin development...

```
git clone -b bug/skull-owner https://github.com/WasabiThumb/xclaim.git
cd xclaim
mvn clean package
```
More or less. Requires git, maven, and [Java 17](https://adoptium.net/temurin/releases/?package=jdk&version=17) (the plugin itself uses Java 8 but newer JDKs will potentially have less problems with dependency resolution). You can also open the project in [IntelliJ IDEA](https://www.jetbrains.com/idea/download) and it will figure out the maven & java parts for you. Once it's fully imported you can use the bar on the right side to run "clean" and then "package". Either way, the plugin JAR will end up in the "target" directory.

**Comment by derjungeinhd** [comment]
Thanks, I had already opened it in Intellij Idea. I didn't know which option does what and I was confused. The bug seems fixed, but there is still no Chat message to type something in the chat, but I try to do a fix for that. Also, I have some fixes for the german translation, since I speak german (visible by my name lol)

**Comment by derjungeinhd** [comment]
So, the initial Issue seems fixed. After my testing, everything worked well.

To the Chat messages, I created a [PR](https://github.com/WasabiThumb/xclaim/pull/74) with a solution to that. It includes also the already mentioned fixes of the german translation. 
Also, I stumbled upon another bug while I was testing: When I put the name "a" in the chat to add it as a player, the process would crash because of no catch for the HTTP 404 code.

I hope my code helps because as I've already mentioned, I'm not really experienced with general plugin development and the general use of Github, so my code and PR could be complete garbage. But thanks for taking time fixing this Issue!
